Fix publish task dependency conflicts, add publishing of iOS artifacts

kvaster opened the pull request:
It seems that gradle becomes crazy with publishing signed artifacts in case there is 'common' artifact for different platform: https://github.com/gradle/gradle/issues/26091#issuecomment-1836156762

Removing javadoc stub fixes the problem.

Also building iOS/macos artifacts is done using separate pipeline on macos runner.

Also we need to update to kotlin 2.1.0-Beta1 cause of folowing problem: https://youtrack.jetbrains.com/issue/KT-70700/Gradle-8.10-The-value-for-task-commonizeNativeDistribution-property-kotlinNativeBundleBuildService-cannot-be-changed-any-further

Fixes: https://github.com/klogging/klogging/issues/294

mjstrasser commented:
Hi @kvaster thanks for this PR. 

I am overseas and will look at it closely next week when I am home again. I hope that is OK.

kvaster commented:
I've also updated PR - I've added publishing of Android 'debug' version, cause in other way you need something like this in your android project to work:

```kotlin
android {
    buildTypes {
        debug {
            matchingFallbacks += listOf("release")
        }
    }
}
```

mjstrasser reviewed (approved):
LGTM
